src/extension.ts

```diff
@@ -109,4 +114,105 @@ export const activate = () => {
         waitUntil(performFixes())
     })
+
+    vscode.workspace.onDidChangeTextDocument(
+        ({ contentChanges, document, reason }) => {
+            if (!getExtensionSetting("insertMissingCommaOnEnter")) {
+                return;
+            }
+
+            if (
+                !vscode.languages.match(['json', 'jsonc'], document)
+            ) {
+                return;
+            }
+
+            if (contentChanges.length === 0) {
+                return;
+            }
+
+            contentChanges = [...contentChanges].sort((a, b) => a.range.start.compareTo(b.range.start))
+
+            const editor = vscode.window.activeTextEditor;
+
+            if (
+                document.uri !== editor?.document.uri ||
+                ["output"].includes(editor.document.uri.scheme)
+            ) {
+                return;
+            }
+
+            if (
+                vscode.workspace.fs.isWritableFileSystem(
+                    document.uri.scheme
+                ) === false
+            ) {
+                return;
+            }
+
+            if (oneOf(reason, vscode.TextDocumentChangeReason.Undo, vscode.TextDocumentChangeReason.Redo)) {
+                return
+            }
+
+            if (contentChanges.some((change) => !isContainEoL(change.text))) {
+                return;
+            }
+
+            let temporaryFileWithoutComments: string;
+
+            const getFileContentWithoutComments = () => {
+                temporaryFileWithoutComments = temporaryFileWithoutComments || stripJsonComments(document.getText(), { whitespace: false });
+
+                return temporaryFileWithoutComments;
+            }
+
+            void editor.edit((edit) => {
+                for (const [i, change] of contentChanges.entries()) {
+                    const prevLinePosition = change.range.start;
+
+                    const prevLine = document.lineAt(prevLinePosition.line + i);
+                    const prevLineText = prevLine.text;
+
+                    if (startsWithComment(prevLine.text.trim())) {
+                        continue;
+                    }
+
+                    const currentLineText = document.lineAt(prevLine.lineNumber + 1).text.trim();
+
+                    const isCurrentLineEmpty =
+                        currentLineText.trim() === "";
+                    const isCurrentLineBeforeComment = startsWithComment(currentLineText);
+
+                    if (!isCurrentLineEmpty && !isCurrentLineBeforeComment) {
+                        continue;
+                    }
+
+                    const fileContentWithoutComments = getFileContentWithoutComments();
+                    const prevLineTextWithoutComments = getTextByLine(fileContentWithoutComments, prevLine.lineNumber)!;
+
+                    const isComment = prevLineTextWithoutComments.trim() !== prevLineText.trim();
+
+                    const prevLineLastChar = isComment ? prevLineTextWithoutComments.trim().at(-1) : prevLineText.at(-1);
```

Comment:
Why don't always grab last character from `prevLineTextWithoutComments`?

Reply:
Did a little refactoring. Please test it

Reply:
Yes, this is much better